crates/turborepo-scm/src/git.rs

```diff
@@ -81,6 +90,71 @@ impl SCM {
     }
 }
 
+const UNKNOWN_SHA: &str = "0000000000000000000000000000000000000000";
+
+#[derive(Debug, Deserialize, Clone)]
+struct GitHubCommit {
+    id: String,
+}
+
+#[derive(Debug, Deserialize, Default)]
+struct GitHubEvent {
+    #[serde(default)]
+    before: String,
+
+    #[serde(default)]
+    commits: Vec<GitHubCommit>,
+}
+
+impl GitHubEvent {
+    fn get_first_commit_id(&self) -> Option<String> {
```

Comment:
Nitpick, but naming is a little misleading here. It's returning `first_commit^` which is really a reference to the _parent_ of the first commit, not the first commit itself.

Reply:
thanks! updated in https://github.com/vercel/turborepo/pull/9169/commits/88e4a498e76adf914f0edae2eb6badabd5ded509